config/config.go

```diff
@@ -701,6 +702,12 @@ func New(v *viper.Viper) (*Configuration, error) {
 	// Migrate combo stored request config to separate stored_reqs and amp stored_reqs configs.
 	resolvedStoredRequestsConfig(&c)
 
+	mergedBidderInfos, err := applyBidderInfoConfigOverrides(c.BidderInfos, bidderInfos)
```

Comment:
Would it make sense to insert the `bidderInfos` loaded from disk into the `config` struct before the viper load step, so viper does the merge automatically? And then perhaps a validate infos step to handle the extra logic in the Overrides function.

Reply:
I tried it and found interesting behavior. In `setBidderDefaults` bidder infos in viper initialized with empty values. If any field for adapter was set through env variables or pbs.json or pbs.yaml file - Viper sets this value to bidder info.
If I set bidder info from file system to config before unmarshalling - Viper replaces everything for bidders that were "extra" configured through env variables. These bidders only have data from env variables and nothing from file system. Bidders that were not configured through env variables have data from file system. 

Hope I understand you correctly, basically you suggest to do something like this: 
```
// New uses viper to get our server configurations.
func New(v *viper.Viper, bidderInfos BidderInfos) (*Configuration, error) {
	var c Configuration
	c.BidderInfos = bidderInfos ---> set loaded bidder info from filesystem here
	if err := v.Unmarshal(&c); err != nil {
		return nil, fmt.Errorf("viper failed to unmarshal app config: %v", err)
	}
      .....
```

Reply:
Could you please add more details about `validate infos step to handle the extra logic in the Overrides function`? Move bidder infos validation to `applyBidderInfoConfigOverrides`?

Reply:
The above is what I was thinking. So you discovered that if, for example, a bidder's `endpoint` was set in an environment variable, viper would override the initial setting with that from the environment variable. However if it was set in `pbs.yaml`, viper would ignore it and keep what was originally set before the unmarshal? If this is true, there is some sort of inconsistency with viper. I don't think it will be worthwhile debugging that inconsistency, my suggestion was mainly in hopes of keeping the code simpler and easier to read, there should be no real functional advantage in doing it this way.

Reply:
Discussed this offline. I didn't find better and easier way to achieve same result using Viper functionality.